### PR TITLE
Add bool support

### LINKED ISSUE #6: Booleans
Implement booleans, such that

```python

@dataclass
class Config:
  b: bool
```

Creates two arguments: `--b` which sets b to `True`, and `--no-b` which sets b to `False`.

### PR DESCRIPTION
**Pull request by gverstegen**
fixes #6

### CODE REVIEW

File: tests/test_parse.py [hunk 1]
```diff
@@ -27,6 +31,30 @@ def test_float(self):
...
+
+    def test_bool_lower(self):
+        c = parse(Config, ["--a", "12", "--f", "true"])
+        assert c.f is True
+
+    def test_bool_upper_false(self):
+        c = parse(Config, ["--a", "12", "--f", "False"])
+        assert c.f is False
+
+    def test_bool_upper_true(self):
```
Comment (antonsteenvoorden) [inline tests/test_parse.py:53]: maybe add 1 missing test case where you enter some gibberish and expect the error

File: tests/test_parse.py [hunk 2]
```diff
@@ -27,6 +31,30 @@ def test_float(self):
         c = parse(Config, ["--a", "12", "--c", "1.23"])
         assert c.c == 1.23
 
-    def test_bool(self):
-        c = parse(Config, ["--a", "12", "--d", "True"])
+    def test_bool_0(self):
```
Comment (rogiervandergeer) [inline tests/test_parse.py:37]: I think it would be nicer to put these tests in a separate class

File: src/pydargs/__init__.py [hunk 3]
```diff
@@ -22,14 +22,48 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None) -> Dataclass:
...
     for field in fields(tp):
-        parser.add_argument(
-            f"--{field.name.replace('_', '-')}",
-            default=field.default,
-            dest=field.name,
-            help=f"Override field {field.name}.",
-            type=field.type,
-        )
+        if field.type is bool:
+            if arg_type := field.metadata.get("bool_arg_type", None):
```
Comment (rogiervandergeer) [inline src/pydargs/__init__.py]: Wouldn't it be nicer to have the flags as default?

I would certainly prefer a metadata field that is a bit more self-explanatory. E.g. `as_flags=True`

Comment (antonsteenvoorden) [reply src/pydargs/__init__.py]: Maybe `add_flags` even?

File: tests/test_parse.py [hunk 4]
```diff
@@ -27,6 +32,36 @@ def test_float(self):
...
+        c = parse(Config, ["--a", "12", "--g"])
+        assert c.g is True
+
+    def test_bool_no_flag(self):
+        c = parse(Config, ["--a", "12", "--no-g"])
+        assert c.g is False
+
+    def test_bool_too_many_arguments(self):
+        with raises(SystemExit):
+            parse(Config, ["--a", "12", "--g", "help"])
```
Comment (antonsteenvoorden) [inline tests/test_parse.py:67]:
```suggestion
            parse(Config, ["--a", "12", "--f", "help"])
```

File: tests/test_parse.py [hunk 5]
```diff
@@ -27,6 +32,36 @@ def test_float(self):
...
+    def test_bool_flag(self):
+        c = parse(Config, ["--a", "12", "--g"])
+        assert c.g is True
+
+    def test_bool_no_flag(self):
+        c = parse(Config, ["--a", "12", "--no-g"])
+        assert c.g is False
+
+    def test_bool_too_many_arguments(self):
+        with raises(SystemExit):
```
Comment (antonsteenvoorden) [inline tests/test_parse.py:66]:
```suggestion
        with raises(ValueError):
```

**Review by rogiervandergeer — APPROVED** [review]

**Pull request merged**